lib/utils/simnet-connext-channels.ts

```diff
@@ -0,0 +1,196 @@
+import { from, of, empty, Observable, combineLatest, throwError } from 'rxjs';
+import { mergeMap, concat, map, catchError, delay, repeat, retryWhen, take } from 'rxjs/operators';
+import { XudClient } from '../proto/xudrpc_grpc_pb';
+import grpc from 'grpc';
+import http from 'http';
+import { loadXudClient } from '../cli/command';
+import {
+  GetBalanceRequest,
+  GetBalanceResponse,
+  GetInfoRequest,
+  GetInfoResponse,
+  OpenChannelRequest,
+  OpenChannelResponse,
+ } from '../proto/xudrpc_pb';
+
+interface GrpcResponse {
+  toObject: Function;
+}
+
+type Balances = {
+  channelBalance: number;
+  walletBalance: number;
+};
+
+const processResponse = (resolve: Function, reject: Function) => {
+  return (error: grpc.ServiceError | null, response: GrpcResponse) => {
+    if (error) {
+      reject(error);
+    } else {
+      const responseObj = response.toObject();
+      resolve(responseObj);
+    }
+  };
+};
+
+const getBalance = async (client: XudClient, currency?: string): Promise<GetBalanceResponse.AsObject> => {
+  const request = new GetBalanceRequest();
+  if (currency) {
+    request.setCurrency(currency.toUpperCase());
+  }
+  const balances = await new Promise((resolve, reject) => {
+    client.getBalance(request, processResponse(resolve, reject));
+  });
+  return balances as GetBalanceResponse.AsObject;
```

Comment:
In general it feels a little cleaner to me to use the native grpc response/request objects rather than converting them to javascript objects. Internally I know we use `toObject` and `AsObject` in a few places but I think mainly that's where we're trying to keep grpc-related code separate from internal code that shouldn't be concerned with how our rpc layer is implemented. But since this tool is explicitly using our grpc interface I think it's appropriate here.

I don't think it applies here but for some of the the data types like `bytes` you have more control over the format/encoding of the fields using the grpc objects vs when you use `AsObject` you get less clear types like `string | Buffer`.

Reply:
> In general it feels a little cleaner to me to use the native grpc response/request objects rather than converting them to javascript objects.

For sure. It's much cleaner now with native grpc objects.